lib/sentry/integrations/oban/error_reporter.ex

```diff
@@ -24,27 +24,43 @@ defmodule Sentry.Integrations.Oban.ErrorReporter do
           :no_config
         ) :: :ok
   def handle_event([:oban, :job, :exception], _measurements, %{job: job} = _metadata, :no_config) do
-    %{reason: exception, stacktrace: stacktrace} = job.unsaved_error
+    %{reason: reason, stacktrace: stacktrace} = job.unsaved_error
 
     stacktrace =
       case {apply(Oban.Worker, :from_string, [job.worker]), stacktrace} do
         {{:ok, atom_worker}, []} -> [{atom_worker, :process, 1, []}]
         _ -> stacktrace
       end
 
-    _ =
-      Sentry.capture_exception(exception,
+    fingerprint_opts =
+      (is_exception(reason) && [inspect(reason.__struct__), Exception.message(reason)]) ||
+        [inspect(reason)]
+
+    opts =
+      [
         stacktrace: stacktrace,
         tags: %{oban_worker: job.worker, oban_queue: job.queue, oban_state: job.state},
-        fingerprint: [
-          inspect(exception.__struct__),
-          inspect(job.worker),
-          Exception.message(exception)
-        ],
+        fingerprint:
+          [
+            inspect(job.worker)
+          ] ++ fingerprint_opts,
```

Comment:
Nit but saves some space 😉 

```suggestion
        fingerprint: [inspect(job.worker)] ++ fingerprint_opts,
```